RadixWallet/Profile/Entity/OnLedgerSettings/Account+OnLedgerSettings.swift

```diff
@@ -54,25 +58,89 @@ extension Profile.Network.Account.OnLedgerSettings {
 		/// Controls the ability of thir-parties to deposit into this account
 		public var depositRule: DepositRule
 
-		/// Denies or allows third-party deposits of specific assets by ignoring the `depositMode`
-		public var assetsExceptionList: OrderedSet<AssetException>
+		/// Denies or allows third-party deposits of specific assets by ignoring the `depositMode`,
+		/// `nil` means that the account was "recovered" using "Account Recovery Scan" features,
+		/// thus the value is unknown.
+		private var assetsExceptionList: OrderedSet<AssetException>?
+
+		public func assetsExceptionSet() -> OrderedSet<AssetException> {
+			assetsExceptionList ?? []
+		}
+
+		public mutating func updateAssetsExceptionList(_ update: (inout OrderedSet<AssetException>?) -> Void) {
+			update(&self.assetsExceptionList)
+		}
+
+		public mutating func setAssetsExceptionList(
+			_ new: OrderedSet<AssetException>?
+		) {
+			updateAssetsExceptionList { $0 = new }
+		}
+
+		public mutating func removeAllAssetsExceptions() {
+			assetsExceptionList = []
+		}
+
+		public mutating func appendToAssetsExceptionList(_ new: AssetException) {
+			if assetsExceptionList == nil {
+				assetsExceptionList = [new]
+			} else {
+				assetsExceptionList!.updateOrAppend(new)
+			}
+		}
 
 		/// Allows certain third-party depositors to deposit assets freely.
-		/// Note: There is no `deny` counterpart for this.
-		public var depositorsAllowList: OrderedSet<DepositorAddress>
+		/// Note: There is no `deny` counterpart for this, `nil` means that the account was
+		/// "recovered" using "Account Recovery Scan" features. thus the value is unknown.
+		private var depositorsAllowList: OrderedSet<DepositorAddress>?
+
+		public func depositorsAllowSet() -> OrderedSet<DepositorAddress> {
+			depositorsAllowList ?? []
+		}
+
+		public mutating func removeAllAllowedDepositors() {
+			depositorsAllowList = []
+		}
+
+		public mutating func appendToDepositorsAllowList(_ new: DepositorAddress) {
+			if depositorsAllowList == nil {
+				depositorsAllowList = [new]
+			} else {
+				depositorsAllowList!.updateOrAppend(new)
+			}
+		}
+
+		public mutating func updateDepositorsAllowList(_ update: (inout OrderedSet<DepositorAddress>?) -> Void) {
+			update(&self.depositorsAllowList)
+		}
+
+		public mutating func setDepositorsAllowList(_ new: OrderedSet<DepositorAddress>?) {
+			updateDepositorsAllowList { $0 = new }
+		}
 
 		public init(
 			depositRule: DepositRule,
-			assetsExceptionList: OrderedSet<AssetException>,
-			depositorsAllowList: OrderedSet<DepositorAddress>
+			assetsExceptionList: OrderedSet<AssetException>?,
```

Comment:
So we make it optional because we want to be able to distinguish the case where we don't have it from the empty case? Perhaps that could just be a separate flag?

Reply:
I dont wanna make too many Profile changes without bumping snapshot version (which we dont wanna do) at this point.